### PR TITLE
Fixed processing of model config file.

### PR DESCRIPTION
**Pull request by nwittler**
**Bug**
Instantiating models from configs did not perform some required checks performed when creating the model interactively.

**Fix**
Both use cases now go through the same checks.

**Misc**
This also required to update some test configs to avoid duplicate names.

### CODE REVIEW

**Review by lazyoracle — APPROVED** [review]
Not waiting for the codeclimate analysis since that seems to have run into errors. Everything else looks good!

**Comment by nwittler** [comment]
Did a tiny bit to reduce complexity in c/model.py. Since a rework is in order for the model and propagation libraries in #99, this is postponed.

**Review by lazyoracle — APPROVED** [review]

File: c3/model.py [hunk 1]
```diff
@@ -61,14 +61,29 @@ def get_ground_state(self) -> tf.constant:
...
+    def __check_drive_connect(self, comp):
+        for connect in comp.connected:
+            try:
+                self.subsystems[connect].drive_line = comp.name
+            except KeyError:
+                raise KeyError(
+                    f"Tried to connect {comp.name}"
+                    f" to non-existent device {self.subsystems[connect].name}."
+                )
+
```
Comment (fedroy) [inline c3/model.py:73]: these codecov warning are useful for adding specific tests

**Review by fedroy — APPROVED** [review]
Merging approved, however we could use the codecov warnings to add tests in a timely manner
